### ISSUE TITLE
When adding a project, "Forgot your password?" link should say "Forgot your account info?"

### ISSUE THREAD

**Issue by romw** [BOINC/boinc#1049]
**Reported by JacobKlein on 10 Mar 41246431 17:46 UTC**
Since the hyperlink takes you to a page called "Forgot your account info?", and the page can be used to determine more than just the password (i.e.: can determine email address)...

The link should say: "Forgot your account info?"

Migrated-From: http://boinc.berkeley.edu/trac/ticket/1073

**Comment by romw** [comment]
**Commented by JacobKlein on 3 Mar 41246441 13:46 UTC**
Note: Don't forget to update localizations where appropriate.

**Comment by Ageless93** [comment]
Small cosmetic change. Change Forgot your password? to Forgot your account info?
This will require all new translations though. Talking about the translations, at present the only language I find where it's "forgot **your** password?" is in English, the rest of the translations make it "forgot password?"

**Comment by AenBleidd** [comment]
@Ageless93, please assign this ticket to me. I'll fix it

**Comment by AenBleidd** [comment]
Fixed in #2197

**Comment by davidpanderson** [comment]
I think "forgot your password?" is better

**Comment by CharlieFenton** [comment]
The original entry by @romw in this issue on February 3, 2015 said:

> Reported by @JacobKlein on 10 Mar 41246431 17:46 UTC
> Since the hyperlink takes you to a page called "Forgot your account info?", and the page can be used to determine more than just the password (i.e.: can determine email address)...
> 
> The link should say: "Forgot your account info?"

It is unclear whether the new wording was suggested by @JacobKlein or by @romw.

So @AenBleidd was simply implementing @JacobKlein's recommendation as (apparently) endorsed by @romw and @Ageless93.

Would "Forgot your login info?" be better?

**Comment by davidpanderson** [comment]
The page it takes you to is essentially a send-new-password-by-email page.
The stuff about finding your authenticator is not relevant in normal use.
So we should change both to "Forgot your password?",
which is something people are used to seeing on login pages.

**Comment by AenBleidd** [comment]
Actually this is a problem of current management of BOINC project:
We have 342 issues. Some of them are very old. And all of them are still open.
@davidpanderson, this issue was created 7 years ago. There was no comments on it that it is not needed.
24 days ago @Ageless93 wrote a comment and assigned this ticket to me.
@davidpanderson, still no comments from you.
How should we know your opinion if you didn't write it?
How much time should past to be clear for us that some ticket can be actually done?
There is no quorum, no plans for releases. Nothing.
Some people made such plans but there is still no comments from you.
You wrote your opinion when the work is done. So you make me think that it was done for nothing.
Please spent a some time to make a review of all open tickets and make some comments to start discussion.
I understand that this work is not so interesting as implementing new features. I understand you. But please try understand my position too: I want to help this project. I do this for free just because I want this project to help many people.
So please help us too and take a little bit more responsibility.

Thank you

**Comment by davidpanderson** [comment]
Are you under the impression that I get paid to maintain BOINC?  I don't.  As far as "take a little bit more responsibility": don't tell me what to do.  Have you looked at https://boinc.berkeley.edu/trac/wiki/AdminRoles ?

**Comment by AenBleidd** [comment]
@davidpanderson I do not care whether you paid or not. 
But if you're not agree with some of the changes that comes from the existing issues why didn't you left some comments before? The issue was open for 7 years. You've never left the comment so I thought that the issue is valid and I can fix it. Now you said that it is not valid.
All I want as a part of the community is to see that you take care of the project.
Saying that the issue that is open for 7 years is not necessary after I made a changes 1 month ago tells opposite.
This issue is just an example.
I sent couple of weeks ago a proposition to add unit-tests. You left no comments. So do you agree with unit-tests or not? I dd not know. But I do not want to have the situation when I'll finish my work and then you'd just tell that this is not necessary and should be done in a different way.
I do not want to waste my time making work for nothing.
I hope you can understand my position.
Thank you

**Comment by AenBleidd** [comment]
@Ageless93, @CharlieFenton, @davidpanderson, 
What should I with this issue?
Should I revert my changes and thus this issue will be closed as 'Won't fix'?

**Comment by CharlieFenton** [comment]
> @Ageless93, @CharlieFenton, @davidpanderson,
> What should I with this issue?
> Should I revert my changes and thus this issue will be closed as 'Won't fix'?


I have no opinion about the wording; I merged it only because there seemed to be no problems doing so. If the consensus is to revert it, I can do that in master. My understanding from @TheAspens is that once it is merged the branch cannot be used anymore:
> 
> once you merge a pull request it stops tracking the branch and it can't be used to merge the branch again. 

@TheAspens: When there is this sort of disagreement, how should a decision be made under the current system?

**Comment by davidpanderson** [comment]
I don't think there's disagreement about the string.  The issue is whether it's my responsibility to review all issues.

**Comment by AenBleidd** [comment]
@davidpanderson,
But you did a review after it was implemented and closed.

Please, understand my position (maybe it is a someone else position too):
I do not want to implement a feature, spend my time and then just found that my time was wasted because you didn't review an initial issue but think that these changes are not needed.

So what should I do in this case? Neither you nor other guys from the committee. What should I do?

**Comment by davidpanderson** [comment]
In this case, revert the change and I'll fix the title of the forgot-password web page.
In general, I should review changes involving user-visible text.

**Comment by AenBleidd** [comment]
Done. #2252 
@davidpanderson,
Please review and merge. Thanks

**Comment by CharlieFenton** [comment]
> I don't think there's disagreement about the string. The issue is whether it's my responsibility to review all issues.

I don't think there is any disagreement about whose responsibility it is to review issues. I believe there is disagreement about the wording of the string: @Rom and @Ageless93 felt it should be "Forgot your account info?" and @davidPAnderson feels it should be "Forgot your password?"

That is what I am asking @TheAspens and other members of the PMC for guidance about: how should disagreements about the proper way to resolve issues be arbitrated now and in the future? 

Although the current issue is quite minor, there will be times when a consensus is not reached on more pressing matters, and we should have a procedure in place to address such situations.

But I have not heard from any of the PMC members other than the parties who have different opinions. I would appreciate a timely response.

**Comment by davidpanderson** [comment]
The misnamed "change account info" web page has some stuff at the bottom about looking up your authenticator in account*.xml, and using that to log in.  This is going to go away; it's too techie, of no use, and it won't work as we start using weak authenticators more.  At that point it's purely Reset Password, and the appropriate link is "Forget your password?"

**Comment by Ageless93** [comment]
Sorry for the late answer, real world issues were taking me up (and will do
again for the next unforeseeable time). On being not available, I know that
@TheAspens is on vacation this week, so don't think you'll get an answer
from him so quickly either.

In any case, the way that this code change has been reverted isn't
following the rules we put down in the governance model or the development
workflow papers. (
https://github.com/BOINC/boinc-policy/blob/master/Governance.md,
http://boinc.berkeley.edu/trac/wiki/Development_Workflow)
So I vetoed the review and merge for the moment, until more committers have
come along and said their bit.

I asked about progress for this issue, because it wasn't progressing in any
direction. One of my 'jobs' as I see it, is to see if Issues have been
looked at by others and if not, see if someone wants to take them on, if
their original poster thinks they're still valid, or if they can be locked
due to out-of-date or other reasons.

Looking into the specifics of this issue, when logging in, a user sees
"Email address: forgot email address?, Password: forgot password?" next to
their login lines.
That'll get you to a page, such as
https://setiathome.berkeley.edu/get_passwd.php
The link to the page already implies that you'll be able to get your
password back. The page itself however is called "Forgot your account
info?" with sub text:
"1) If you know your account's email address, and you can receive email
there:
Enter the email address below, and click OK. You will be sent email
instructions for resetting your password."

and

"2) If you forgot your account's email address, or you can't receive email
there:
Account key instructions + You will now be logged in to your account;
update the email and password of your account."

Everything in the text points to reset of password, not of account info,
which in my point of view is more general: nickname, email address,
password, other account info, URL, Profile, team., etc. Now, granted, all
of that can be changed from there, but that's not what we imply with the
"forgot password?" link and text on that page.

So I think the page should be more in line with what we imply, and the text
for "Forgot your account info?" should be "Forgot your password?", and if
that was the original setup, that it should be reverted. But then I checked
other projects with older pages and I find it differs:
https://www.primegrid.com/get_passwd.php doesn't have any such text (which
may be a good idea in this?)
https://www.cpdn.org/cpdnboinc/get_passwd.php has "Forgot your account
info?" and they haven't updated in a while.
http://www.enigmaathome.net/get_passwd.php is anicent and it has "Forgot
your account info?"

Which makes me wonder how the change from "Forgot your password?" ever
happened in the direction of "Forgot your account info?" in this Issue and
its PRs.
While the original Issue asked for the LINK to change to
forgot_account_info:
> Since the hyperlink takes you to a page called "Forgot your account
info?", and the page can be used to determine more than just the password
(i.e.: can determine email address)...

> The link should say: "Forgot your account info?"



-- Jord van der Elst.

**Comment by davidpanderson** [comment]
everyone involved agrees now.  Please un-veto.

**Comment by Ageless93** [comment]
Charlie doesn't agree. He's even asking that @TheAspens and/or the PMC look in (although this isn't something for them to decide upon).
I don't agree, with reasons given above. I'd like them answered first. 

I also feel that @AenBleidd only changed things around because he felt irritated with how things had run their course. A whole discussion beforehand on the subject, then changes made and only after the fact and after the change, did you say you didn't agree. There was no real consensus in the change decision, it was just you who disagreed, Charlie who asked good questions and Vitalii who hastily made the code change to be rid of it. 

I don't feel that this requires a hasty change, it's not something that breaks things if it isn't added overnight, so with all due respect, I'd like to wait for more input from others. At least from @TheAspens when he's back from vacation.

**Comment by AenBleidd** [comment]
To be honest, I'mm too tired to fight with this anymore.
In my country we have a phraseme: "You're the boss I'm a fool".
So I prefer to be the second one in this case.

**Comment by CharlieFenton** [comment]
I neither agree nor disagree. I have no objections to either wording. I only inquired because I wanted to understand the general process for resolving disagreements about code implementation, since @Rom and @Ageless93 apparently supported the change to "Forgot your account info?" and @davidPAnderson feels it should be "Forgot your password?"

I did not intend to blow this minor issue out of proportion.

**Comment by CharlieFenton** [comment]
> So I vetoed the review and merge for the moment, until more committers have
come along and said their bit.

@Ageless93: 7 days have passed and no one has added any more comments. Does PR #2254 satisfy your other objections? Are you ready to remove your veto?

**Comment by tristanolive** [comment]
The development process was followed as far as code review is concerned; a contribution was made via a pull request and there was plenty of time for review, then a committer signed off on it and merged the PR. Now that we're after that point, a new PR can be made to change the code again, if someone realizes that there is a new problem to address or a better way to do it.

In the future, we can likely avoid problems with these sorts of very old issues by using the [appropriate project board](https://github.com/BOINC/boinc/projects/4) to ensure the workflow is followed. A 7-year-old issue should never jump from the backlog straight into "Ready to implement". It should first go into the "Review" category so that any needed discussion can take place prior to development time being wasted on something that is obsolete or incorrectly designed. This does require committers to be in the habit of checking for issues in need of review (and likely discussing some of them on regular development calls), so it will take some time to get used to the process, but that sure is better than anyone being expected to review hundreds of backlogged issues!
